src/dpos-user.ts

```diff
@@ -79,6 +81,52 @@ export class DPOSUser {
     )
   }
 
+  static async createEthSignMetamaskUserAsync(
+    web3: Web3,
+    dappchainEndpoint: string,
+    dappchainKey: string,
+    chainId: string,
+    gatewayAddress: string,
+    loomAddress: string
+  ): Promise<DPOSUser> {
+    const provider = new ethers.providers.Web3Provider(web3.currentProvider)
+    const wallet = provider.getSigner()
+
+    const { client, address } = createDefaultEthSignClient(
+      dappchainKey,
+      dappchainEndpoint,
+      chainId,
+      wallet
+    )
+
+    const ethAddress = await wallet.getAddress()
+    const dappchainLoom = await Coin.createAsync(
+      client,
+      new Address('eth', LocalAddress.fromHexString(ethAddress))
+    )
+
+    const dappchainDPOS = await DPOS2.createAsync(
+      client,
+      new Address('eth', LocalAddress.fromHexString(ethAddress))
+    )
+
+    const dappchainGateway = await LoomCoinTransferGateway.createAsync(client, address)
+    const dappchainMapper = await AddressMapper.createAsync(client, address)
```

Comment:
This doesn't look right, should use the `Address('eth', LocalAddress.fromHexString(ethAddress))` here too instead of `address`?

Reply:
Not sure, the address mapper responsible for a map from loomchain address to eth address should have the caller using the loomchain address

Reply:
I think you're right about the AddressMapper caller. The Gateway should probably take the eth address though.

However, the entire point of this "sign with your Ethereum key" feature is to avoid dealing with the DAppChain key as much as possible, but this function expects a DAppChain key... I think a better way to do this is to assume the account mapping already exists.

Reply:
When the address mapper exists, the address used on the contracts is the dappchain address, in this case for the dpos the address attached to tokens is the dappchain address, so is a case when the mapping exists